### ISSUE TITLE
Overlapping overload detection fails when signature contains TypeVar

### ISSUE THREAD

**Issue by erictraut** [python/mypy#10157]
Mypy detects and reports overlapping function overloads that differ in return type.

In the example below, it correctly reports that the second overload of `func1` overlaps the first overload. However, when the concrete type `str` is replaced by a generic type `_T`, it fails to report the error.

```python
@overload
def func1(
    a: str,
    b: Literal[False] = ...,
) -> List[str]: ...
@overload
def func1(
    a: str,
    b: bool = ...,
) -> List[Union[str, int]]: ...

_T = TypeVar("_T")
@overload
def func2(
    a: _T,
    b: Literal[False] = ...,
) -> List[_T]: ...
@overload
def func2(
    a: _T,
    b: bool = ...,
) -> List[Union[_T, int]]: ...
```

**Comment by JukkaL** [comment]
This is a false negative. Both overloaded functions are overlapping.
